spec/factories/scheduled_event.rb

```diff
@@ -2,9 +2,50 @@
   factory :scheduled_event do
   	type 'holiday'
   	event_name 'Christmas'
-  	start_time { Date.new(2017, 1, 1) }
-  	one_time true
-  	recurring_rules nil
-  	offset_rule '3'
-  end
+    type 'holiday'
+    event_name 'Christmas'
+    start_time { Date.new(2017, 1, 1) }
+    one_time true
+    offset_rule 3
+
+	  trait :empty_recurring_rules do
+	    recurring_rules nil
+	  end
+
+	  trait :recurring_rules do
+	   recurring_rules { }  
+	  end
+
+	  trait :offset_0 do
```

Comment:
I don't think these extra traits are necessary... you can just assign that as a value when you create your mocks in the test. 
```
FactoryGirl.create(:scheduled_event, offset_rule: 3) 
```
vs
```
FactoryGirl.create(:scheduled_event, :offset_3) 
```
doesn't really gain much